src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowNode.cs

```diff
@@ -5,16 +5,17 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using Newtonsoft.Json;
 
 namespace Microsoft.DotNet.DarcLib
 {
     public class DependencyFlowNode
     {
-        public DependencyFlowNode(string repository, string branch)
+        public DependencyFlowNode(string repository, string branch, string id)
         {
             Repository = repository;
             Branch = branch;
-            Id = Guid.NewGuid().ToString();
+            Id = id;
```

Comment:
Is this Id actually needed?

Reply:
I am using Id in the BarViz code. It made it much easier/cleaner to determine what nodes the edges connect. I added it to DependencyFlowNode to decrease the number of times I had to walk the graph (rather than doing it in the conversion from DependencyFlowGraph to the json FlowGraph that is generated by the ChannelsController). Basically, I didn't want to have to pass along the full node information in the Edge object, and only wanted to pass along some sort of Id.